src/core/MeshBVH.js

```diff
@@ -943,13 +964,27 @@ export class MeshBVH {
 
 		);
 
-		return Math.sqrt( closestDistanceSq );
+		const closestDistance = Math.sqrt( closestDistanceSq );
 
-	}
+		if ( target ) {
+
+			if ( ! target.point ) target.point = temp1.clone();
+			else target.point.copy( temp1 );
+			target.distance = closestDistance,
+			target.faceIndex = closestDistanceTriIndex;
+			return target;
+
+		} else {
```

Comment:
I think it's okay to default `target` to an empty object in the constructor which will simplify final condition block further, as well. Then we won't have to handle the "else" condition because it would be guaranteed that the target object is defined.

We can do the same for `closestPointToGeometry` for at least `target1`.